FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -689,17 +689,17 @@ - (void)setAPNSToken:(NSData *)APNSToken withUserInfo:(NSDictionary *)userInfo {
     return;
   }
   // Use this token type for when we have to automatically fetch tokens in the future
-#if TARGET_OS_SIMULATOR && TARGET_OS_IOS
+#if TARGET_OS_SIMULATOR
```

Comment:
Did you find evidence that APNS is supported on the tvOS and visionOS simulators?

Reply:
I haven't unfortunately (but I also haven't looked or tried it yet). In the two cases of `TARGET_OS_SIMULATOR` I found, it seemed reasonable to assume it's a sandbox (not production) app given that it's a simulator, regardless of platform.